Containerfile

```diff
@@ -1,28 +1,24 @@
-FROM registry.access.redhat.com/ubi7/ubi AS builder
-ENV SEARCHD_RPM=sphinx-2.2.11-1.rhel7.x86_64.rpm \
-    SEARCHD_RPM_SHA=959b04eb3f7fb2314d7a2702b61e9b3e627b66b1a8574dece21c0592be1b90e2 \
-    SEARCHD_SRC=sphinx-2.2.11-release.tar.gz \
-    SEARCHD_SRC_DIR=sphinx-2.2.11-release \
-    SEARCHD_SRC_SHA=6662039f093314f896950519fa781bc87610f926f64b3d349229002f06ac41a9
+FROM quay.io/centos/centos:stream9 AS builder
+ENV SEARCHD_REF=6.2.12 \
+    SEARCHD_REPO=https://github.com/manticoresoftware/manticoresearch.git \
+    CC=clang-16 \
+    CXX=clang++-16 \
+    BUILD_PATH=/tmp/manticore_uselessly_long_path_to_prevent_rpm_build_issues \
+    BUILD_FLAGS="-DUSE_SYSLOG=0 -DWITH_GALERA=0 -DWITH_RE2=0 -DWITH_STEMMER=0 -DWITH_ICU=1 -DWITH_SSL=1 -DWITH_ZLIB=1 -DWITH_ODBC=0 -DWITH_EXPAT=0 -DWITH_ICONV=1 -DWITH_POSTGRESQL=0 -DWITH_MYSQL=0 -DBUILD_TESTING=0"
+WORKDIR $BUILD_PATH
+
 SHELL ["/bin/bash", "-x", "-o", "pipefail", "-c"]
+# clang is broken on s390x (RHEL-15874), to use gcc remove CC and CXX variables,
+#   also replace llvm-toolset with make automake gcc gcc-c++ kernel-devel
 # hadolint ignore=DL3003,DL3032,SC2046
-RUN if [ "$(uname -m)" == "x86_64" ]; then \
-      yumdownloader --downloadonly --destdir=/home/rpms mariadb-libs.$(uname -m) postgresql-libs.$(uname -m) unixODBC.$(uname -m) --resolve && \
-      curl -sSL https://sphinxsearch.com/files/$SEARCHD_RPM -o /home/rpms/$SEARCHD_RPM && \
-     sha256sum -c - <<< "$SEARCHD_RPM_SHA /home/rpms/$SEARCHD_RPM" && \
-     mkdir /home/searchd ; \
-    else \
-      yumdownloader --downloadonly --destdir=/home/rpms mariadb-libs.$(uname -m) --resolve && \
-      yum install -y gcc-c++ mysql-devel make  && \
-      curl -sSL http://sphinxsearch.com/files/$SEARCHD_SRC -o /tmp/$SEARCHD_SRC && \
-      sha256sum -c - <<< "$SEARCHD_SRC_SHA /tmp/$SEARCHD_SRC" && \
-      tar xzf /tmp/$SEARCHD_SRC -C /tmp && \
-      cd /tmp/$SEARCHD_SRC_DIR && ./configure && DESTDIR=/home/searchd make install && \
-      rm -rf /tmp/sphinx* ; \
-    fi
-
+RUN yum install -y --setopt=skip_missing_names_on_install=False,tsflags=nodocs llvm-toolset mysql cmake boost-devel openssl-devel zlib-devel bison flex systemd-units rpm-build git && \
+      git clone --depth=1 --branch=$SEARCHD_REF $SEARCHD_REPO . && \
+      sed -i -e 's/Boost_USE_STATIC_LIBS ON/Boost_USE_STATIC_LIBS OFF/' src/CMakeLists.txt && \
```

Comment:
What is this for? I think it deserves a comment.

Reply:
Well, compilation fails without it. I think it is a variable to allow dynamic binding to the boost library. Because in rhel it only comes as a dynamic lib, not as a static lib `.a`.

But I'm bot sure you can add a comment in the middle of a RUN command. Also not sure comment will be very useful. Maybe suggest a comment line that will make things more clear for you as a reader?

Reply:
Added a comment, let me know if it makes things clear enough

Reply:
Looks good!